mnemonic-memory-services/mnemonic-pmdk-vmem-service/src/main/java/org/apache/mnemonic/service/memory/internal/PMDKVMemServiceImpl.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.mnemonic.service.memory.internal;
+
+import org.apache.mnemonic.query.memory.EntityInfo;
+import org.apache.mnemonic.query.memory.ResultSet;
+import org.apache.mnemonic.service.computing.ValueInfo;
+import org.apache.mnemonic.service.memory.MemoryServiceFeature;
+import org.apache.mnemonic.service.memory.VolatileMemoryAllocatorService;
+import org.flowcomputing.commons.primitives.NativeLibraryLoader;
+import org.flowcomputing.commons.resgc.ReclaimContext;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.HashSet;
+
+public class PMDKVMemServiceImpl implements VolatileMemoryAllocatorService {
+  private static boolean nativeLoaded = false;
+  static void loadNativeLibrary() {
+    try {
+      NativeLibraryLoader.loadFromJar("pmdkvmemallocator");
+    } catch (Exception e) {
+      throw new Error(e);
+    }
+    nativeLoaded = true;
+  }
+
+  protected Map<Long, Long> m_info = Collections.synchronizedMap(new HashMap<Long, Long>());
+
+  @Override
+  public String getServiceId() {
+    return "pmdk_vmem";
+  }
+
+  @Override
+  public long init(long capacity, String uri, boolean isnew) {
+    if (!nativeLoaded) {
+      loadNativeLibrary();
+    }
+    long ret = ninit(capacity, uri, isnew);
+    m_info.put(ret, capacity);
+    return ret;
+  }
+
+  @Override
+  public long adjustCapacity(long id, long reserve) {
```

Comment:
It s better to add java documentation for these class methods or inherit from the super class.

Reply:
sure, will add it.

Reply:
Looks Javadoc will duplicate the overriding method comment from its super class as default.